app/uk/gov/hmrc/cdsreimbursementclaimfrontend/models/Nonce.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cdsreimbursementclaimfrontend.models
+
+import play.api.libs.json.Format
+import java.util.Base64
+import java.nio.charset.StandardCharsets
+import scala.util.Random
+import scala.util.Try
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.utils.SimpleDecimalFormat
+
+/** Random integer value container. */
+sealed trait Nonce {
```

Comment:
What is Nonce? How come it is different from checksum?

Reply:
[nonce](https://en.wikipedia.org/wiki/Cryptographic_nonce)

Reply:
in our case, we will use nonce when communicating with the upload microservice